crates/bevy_ecs/src/world/mod.rs

```diff
@@ -1300,6 +1300,18 @@ impl World {
             .map(Into::into)
     }
 
+    /// Registers a new [`Resource`] type and returns the [`ComponentId`] created for it.
+    ///
+    /// This enables the dynamic registration of new resource definitions at runtime for advanced
+    /// use cases.
```

Comment:
```suggestion
    /// use cases.
    /// Even if you register a `Resource`, there will be no value for this Resource, use [`World::init_resource`] or [`World::insert_resource`] if you need its value in World
```

Probably a good Idea to remind the user that this doesn't insert the Resource in the world. Just a suggestion though, if you have a better wording, use that instead :)